src/ui/components/focus/AutoFocusGuide.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { Platform, StyleProp, ViewStyle } from 'react-native';
+
+export interface AutoFocusGuideProps {
+  /**
+   * Overrides for the style of the guide.
+   */
+  style?: StyleProp<ViewStyle>;
+}
+
+const FOCUS_GUIDE_STYLE: ViewStyle = {
+  width: '100%',
+  justifyContent: 'center',
+  alignItems: 'center',
+};
+
+/**
+ * A TV platform FocusGuide with autofocus capabilities
+ */
+export const AutoFocusGuide = (props: React.PropsWithChildren<AutoFocusGuideProps>) => {
+  const [tvFocusGuideAvailable, setTvFocusGuideAvailable] = useState(true);
+  const { style, children } = props;
+
+  if (Platform.OS === 'ios' && Platform.isTV) {
```

Comment:
Some more optimizations:
```suggestion
  const [tvFocusGuideAvailable, setTvFocusGuideAvailable] = useState(Platform.OS === 'ios' && Platform.isTV);
  const { style, children } = props;

  if (tvFocusGuideAvailable) {
```

Reply:
What I try to achieve here is to run some code when it is available through an installed react-native-tvos package (a special flavour of react-native with additional tv capabilities, like FocusGuides). You could say that on tvOS a user _has to_ use react-native-tvos, so checking for 'ios' + 'tv' should be enough to assume that react-native-tvos is in use. 
User that are only targeting mobile devices and are using the 'regular' react-native. should not run into build errors caused by react-native-tvos specific components.

Reply:
I've reworked this using [`lazy`](https://react.dev/reference/react/lazy), check it out. 😉